index.ts

```diff
@@ -0,0 +1,237 @@
+		return 'Function';
+	}
+
+	if (Array.isArray(value)) {
+		return 'Array';
+	}
+
+	if (Buffer.isBuffer(value)) {
+		return 'Buffer';
+	}
+
+	const tagType = getObjectType(value);
+	if (tagType) {
+		return tagType;
+	}
+
+	if (value instanceof String || value instanceof Boolean || value instanceof Number) {
+		throw new TypeError('Please don\'t use object wrappers for primitive types');
+	}
+
+	return 'Object';
+};
+
+namespace is {
+	export const undefined = isOfType('undefined');
+	export const string = isOfType('string');
+	export const number = isOfType('number');
+	export const boolean = (x: any) => x === true || x === false;
+	export const symbol = isOfType('symbol');
+	export const func = isOfType('function');
+	export const null_ = (x: any) => x === null;
+
+	export const array = Array.isArray;
+	export const buffer = Buffer.isBuffer;
+
+	const isObject = (x: any) => typeof x === 'object';
+
+	export const object = (x: any) => !is.nullOrUndefined(x) && (is.func(x) || isObject(x));
+
+	export const nativePromise = isObjectOfType('Promise');
+
+	const hasPromiseAPI = (x: any) =>
+		!is.null_(x) &&
+		isObject(x) &&
+		is.func(x.then) &&
+		is.func(x.catch);
+
+	export const promise = (x: any) => is.nativePromise(x) || hasPromiseAPI(x);
+
+	export const generator = (x: any) => is.iterable(x) && is.func(x.next) && is.func(x.throw);
+
+	// TODO: Change to use `isObjectOfType` once Node.js 6 or higher is targeted
+	const isFunctionOfType = (type: string) => (x: any) => is.func(x) && is.func(x.constructor) && x.constructor.name === type;
+
+	export const generatorFunction = isFunctionOfType('GeneratorFunction');
+	export const asyncFunction = isFunctionOfType('AsyncFunction');
+
+	export const regExp = isObjectOfType('RegExp');
+	export const date = isObjectOfType('Date');
+	export const error = isObjectOfType('Error');
+	export const map = isObjectOfType('Map');
+	export const set = isObjectOfType('Set');
+	export const weakMap = isObjectOfType('WeakMap');
+	export const weakSet = isObjectOfType('WeakSet');
+
+	export const int8Array = isObjectOfType('Int8Array');
+	export const uint8Array = isObjectOfType('Uint8Array');
+	export const uint8ClampedArray = isObjectOfType('Uint8ClampedArray');
+	export const int16Array = isObjectOfType('Int16Array');
+	export const uint16Array = isObjectOfType('Uint16Array');
+	export const int32Array = isObjectOfType('Int32Array');
+	export const uint32Array = isObjectOfType('Uint32Array');
+	export const float32Array = isObjectOfType('Float32Array');
+	export const float64Array = isObjectOfType('Float64Array');
+
+	export const arrayBuffer = isObjectOfType('ArrayBuffer');
+	export const sharedArrayBuffer = isObjectOfType('SharedArrayBuffer');
+
+	export const truthy = (x: any) => !!x; // eslint-disable-line no-implicit-coercion
+	export const falsy = (x: any) => !x;
+
+	// the tests are designed for Number.isNaN rather than global isNaN
+	export const nan = (x: any) => is.number(x) && isNaN(Number(x)); 
```

Comment:
Why did you change this from the original?

Reply:
Number.isNaN is currently not supported and isNaN() is typeguarded to only accept number
see https://github.com/Microsoft/TypeScript/issues/15149

Reply:
Ok. Maybe you can use `// @ts-ignore` on it? https://blogs.msdn.microsoft.com/typescript/2017/10/31/announcing-typescript-2-6/

If not, at least link to the TS issue in a comment above.